org.eclipse.m2e.feature/feature.xml

```diff
@@ -26,7 +26,33 @@
          id="org.eclipse.m2e.archetype.common"
          download-size="0"
          install-size="0"
-         version="0.0.0"/>
+         version="0.0.0"
+         fragment="true"
+         unpack="false"/>
+
+   <plugin
+         id="org.eclipse.m2e.archetype.catalog"
+         download-size="0"
+         install-size="0"
+         version="0.0.0"
+         fragment="true"
+         unpack="false"/>
+
+   <plugin
+         id="org.eclipse.m2e.archetype.descriptor"
+         download-size="0"
+         install-size="0"
+         version="0.0.0"
+         fragment="true"
+         unpack="false"/>
+
+   <plugin
+         id="org.eclipse.m2e.archetype.maven-artifact-transfer"
+         download-size="0"
+         install-size="0"
+         version="0.0.0"
+         fragment="true"
+         unpack="false"/>
```

Comment:
@laeubi do you see any reason that speaks against including the archetype bundles in this feature?
I would prefer it over the p2.inf file since it is less magic.

Reply:
The p2.inf ensures that all fragments are always installed together.

Reply:
> The p2.inf ensures that all fragments are always installed together (especially for cases like tests).

This can also be achieved by corresponding Import-Packages and is less magic and even less to type.

Reply:
No this can't work because a fragment is always optional, so a consumer would need to import a package from each fragment even if it is never used.

Reply:
> No this can't work because a fragment is always optional, so a consumer would need to import a package from each fragment even if it is never used.

That's right, a fragment is optional but m2e.core.ui already needs to import a package from each of the archetype artifacts (if one does not use require-bundle). The only import that would not be necessary from a class-loading/compiling point of view is the maven-transfer. But I think one not so necessary import is IMHO ok. And just to save that I wouldn't add the p2.inf.